lib/internal/http2/core.js

```diff
@@ -1306,14 +1446,26 @@ class Http2Stream extends Duplex {
   }
 
   _write(data, encoding, cb) {
-    if (this[kID] === undefined) {
+    // When the Http2Stream is first created, it is corked until the
+    // handle and the stream ID is assigned. However, if the user calls
+    // uncork() before that happens, the Duplex will attempt to pass
+    // writes through. Those need to be queued up here.
+    if (this.pending) {
       this.once('ready', this._write.bind(this, data, encoding, cb));
       return;
     }
 
-    this[kUpdateTimer]();
+    // If the stream has been destroyed, there's nothing else we can do
+    // because the handle has been destroyed. This should only be an
+    // issue if a write occurs before the 'ready' event in the case where
+    // the duplex is uncorked before the stream is ready to go. In that
+    // case, drop the data on the floor. An error should have already been
+    // emitted.
+    if (this.destroyed)
+      return;
```

Comment:
this should not be needed, why it is?

Reply:
This is purely defensive. It *shouldn't* ever happen but we still need to verify that it can't ever happen.